sklearn/ensemble/_gradient_boosting.pyx

```diff
@@ -99,7 +99,7 @@ cdef void _predict_regression_tree_inplace_fast(DTYPE_t *X,
 
 @cython.nonecheck(False)
 def predict_stages(np.ndarray[object, ndim=2] estimators,
-                   np.ndarray[DTYPE_t, ndim=2, mode='c'] X, double scale,
+                   np.ndarray[DTYPE_t, ndim=2] X, double scale,
```

Comment:
Not sure whether there was a good reason to have `mode='c'` originally for example for performance reasons. Removing it does fix the bug with warm_start=True when X is a Fortran array.